### PR TITLE
Security: Added user setting for extension policies

### PR DESCRIPTION
**Pull request by abist**
Added a user setting called `extensionsPolicy` which can have 3 values - `allowAll`, `allowNone` and `allowMicrosoft` and their functions are what the names suggest. Fixes https://github.com/Microsoft/sqlopsstudio/issues/2418.

### CODE REVIEW

File: src/vs/workbench/parts/extensions/electron-browser/extensionsActions.ts [hunk 1]
```diff
@@ -2656,33 +2657,44 @@ export class InstallVSIXAction extends Action {
...
-			filters: [{ name: 'VSIX Extensions', extensions: ['vsix'] }],
-			properties: ['openFile'],
-			buttonLabel: mnemonicButtonLabel(localize({ key: 'installButton', comment: ['&& denotes a mnemonic'] }, "&&Install"))
-		}).then(result => {
-			if (!result) {
-				return TPromise.as(null);
-			}
+		// {{SQL CARBON EDIT}}
+		let extensionPolicy = this.configurationService.getValue<string>(ExtensionsPolicyKey);
+		if (extensionPolicy !== ExtensionsPolicy.allowNone) {
```
Comment (MattIrv) [inline src/vs/workbench/parts/extensions/electron-browser/extensionsActions.ts]: Here the code checks if the extension policy is not `allowNone` but in the next file it checks if the extension policy is `allowAll`. Should "Install from VSIX" be enabled when the policy is set to Microsoft only? Those should all get auto installed so I think we can completely disable "Install from VSIX" in that case right?

Comment (abist) [reply src/vs/workbench/parts/extensions/electron-browser/extensionsActions.ts]: "Install from VSIX" is only enabled for "allowAll". The initial check for "allowNone" is to suppress the dialog to pick an extension if extensions aren't allowed at all.

Comment (MattIrv) [reply src/vs/workbench/parts/extensions/electron-browser/extensionsActions.ts]: This code is in the "install from vsix" action though

**Comment by kburtram** [comment]
@abist I'll merge this now to include in today's insider build.

**Pull request merged**